godot-core/src/builtin/quaternion.rs

```diff
@@ -130,70 +138,115 @@ impl Quaternion {
         Quaternion::new(v.x, v.y, v.z, 0.0)
     }
 
+    /// Normalizes the quaternion.
+    ///
+    /// # Panics
+    /// If the quaternion has length of 0.
     pub fn normalized(self) -> Self {
-        self / self.length()
+        let length = self.length();
+        assert!(!length.approx_eq(&0.0), "Quaternion has length 0");
+        self / length
     }
 
+    /// Performs slerp
+    ///
+    /// # Panics
+    /// If either quaternion is not normalized
```

Comment:
"Performs slerp" has exactly 0 information that's not already in the method name 😉 
I would just use the full term, that's enough.

Please also end both the brief description and the panic note with a fullstop.